akka-actor/src/main/scala/akka/actor/ActorSystem.scala

```diff
@@ -268,8 +268,9 @@ object ActorSystem {
     val cl = bootstrapSettings.flatMap(_.classLoader).getOrElse(findClassLoader())
     val appConfig = bootstrapSettings.flatMap(_.config).getOrElse(ConfigFactory.load(cl))
     val defaultEC = bootstrapSettings.flatMap(_.defaultExecutionContext)
+    val uid = ThreadLocalRandom.current.nextLong()
 
-    new ActorSystemImpl(name, appConfig, cl, defaultEC, None, setup).start()
+    new ActorSystemImpl(name, appConfig, cl, defaultEC, None, setup, uid).start()
```

Comment:
constructor parameter for testing purposes? given my other comment regarding compatibility for classic remoting it might be easier to just create the uid inside the ActorSystemImpl

Reply:
my main motivation was to avoid it being `0L` during ActorSystem object construction: if (e.g. after a refactor) there'd be any fields initialized before the `uid` field, and that initialization would produce logging, it would read the uid before it'd be set.

Perhaps that's over-thinking it though, I'll just make sure it's the first initialization.